tools/serve/serve.py

```diff
@@ -543,23 +543,19 @@ def create_daemon(self, init_func, host, port, paths, routes, bind_address,
 
         if self.daemon:
             try:
-                self.daemon.start(block=False)
+                self.daemon.start()
                 try:
-                    self.stop.wait()
+                    self.stop_flag.wait()
+                    self.daemon.stop()
```

Comment:
Should we have this in a `finally` block instead, in case the above throws (somehow)?